src/main/java/seedu/address/logic/commands/EditExamCommand.java

```diff
@@ -78,14 +78,18 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_EXAM_NOT_EDITED);
         }
 
+        boolean isEditedExamOfSameModule = examToEdit.getModule().isSameModule(editedExam.getModule());
+
         try {
-            if (!examToEdit.getModule().isSameModule(editedExam.getModule())) {
+            if (!isEditedExamOfSameModule && model.isExamLinkedToTask(examToEdit)) {
+                model.replaceExam(examToEdit, editedExam, false);
                 model.unlinkTasksFromExam(examToEdit);
+                return new CommandResult(String.format(MESSAGE_EDIT_EXAM_SUCCESS, editedExam) + "\n"
+                        + "All the tasks previously linked to this exam are now unlinked.");
             } else {
+                model.replaceExam(examToEdit, editedExam, false);
```

Comment:
This line of code (line 90) seems to be repeated as line 85 has the same line of code. Perhaps can put this code under just under the beginning of the try block line, since regardless of the conditional path taken, this code is executed.

Reply:
Okayy